resource-management/pkg/store/redis/redis_test.go

```diff
@@ -199,3 +202,142 @@ func TestPersistVirtualNodesAssignments(t *testing.T) {
+	GR = NewRedisClient()
+
+	// Start first test
+	t.Log("\nFirst test is starting")
+	testCases := make([]*types.LogicalNode, 2)
+
+	var testCase0 = &types.LogicalNode{
+		Id:              "0002",
+		ResourceVersion: "0003",
+		GeoInfo: types.NodeGeoInfo{
+			Region:            1001,
+			ResourcePartition: 1001,
+		},
+		Conditions:  255,
+		Reserved:    true,
+		MachineType: "machineType2",
+	}
+
+	testCases[0] = testCase0
+
+	var testCase1 = &types.LogicalNode{
+		Id:              "0003",
+		ResourceVersion: "0004",
+		GeoInfo: types.NodeGeoInfo{
+			Region:            1002,
+			ResourcePartition: 1002,
+		},
+		Conditions:  255,
+		Reserved:    true,
+		MachineType: "machineType3",
+	}
+
+	testCases[1] = testCase1
+
+	success := GR.PersistNodes(testCases)
+
+	if success == false {
+		t.Error("Store 2 new Logical Nodes of testCases into Redis in failure")
+	}
+
+	// Set batchLength = 2
+	var batchLength uint64 = 2
+	logicalNodes := GR.BatchLogicalNodesInquiry(batchLength)
+	actualLength := len(logicalNodes)
+
+	assert.NotEqual(t, 0, actualLength)
+	assert.Equal(t, uint64(batchLength), uint64(actualLength))
+
+	for i := 0; i < actualLength; i++ {
+		t.Logf("Index #(%v): ", i)
+		t.Log("=================")
+		t.Logf("Id:        (%v)", logicalNodes[i].Id)
+		t.Logf("Region:    (%v)", logicalNodes[i].GeoInfo.Region)
+		t.Logf("RP:        (%v)", logicalNodes[i].GeoInfo.ResourcePartition)
+	}
+	t.Log("\nFirst test is OK\n")
+
+	// Start second test
+	t.Log("\nSecond test is starting")
+	testNumber := 1000
+	testCases10 := make([]*types.LogicalNode, testNumber)
+
+	startID := 1000001
+	startRV := 10001
+	var startRegion types.RegionName = 10001
+	var startRP types.ResourcePartitionName = 10001
+
+	for i := 0; i < testNumber; i++ {
+		testcase := &types.LogicalNode{
+			Id:              strconv.Itoa(startID),
+			ResourceVersion: strconv.Itoa(startRV),
+			GeoInfo: types.NodeGeoInfo{
+				Region:            startRegion,
+				ResourcePartition: startRP,
+			},
+			Conditions:  255,
+			Reserved:    true,
+			MachineType: "machineType" + "strconv.Itoa(i)",
+		}
+
+		testCases10[i] = testcase
+		startID++
+		startRV++
+		startRegion++
+		startRP++
+	}
+
+	success = GR.PersistNodes(testCases10)
+	if success == false {
+		t.Errorf("Store (%v) new Logical Nodes of testCases into Redis in failure", testNumber)
+	}
+
+	// Set different batchLength = 5
+	batchLength = 5
+	logicalNodes = GR.BatchLogicalNodesInquiry(batchLength)
+	actualLength = len(logicalNodes)
+
+	assert.NotEqual(t, 0, actualLength)
+	assert.Equal(t, uint64(batchLength), uint64(actualLength))
+	t.Log("\nSecond test is OK\n")
+
+	// Start third test
+	t.Log("\nThird test is starting")
+	// Set different batchLength = 1000
+	batchLength = 1000
+	logicalNodes = GR.BatchLogicalNodesInquiry(batchLength)
+	actualLength = len(logicalNodes)
+
+	assert.NotEqual(t, 0, actualLength)
+	assert.Equal(t, uint64(batchLength), uint64(actualLength))
```

Comment:
Why not assert.Equal(t, batchLength, actualLength)?

Reply:
batchLength is int64 type. So change of "assert.Equal(t, batchLength, int64(actualLength))" is working.